gwpy/cli/cliproduct.py

```diff
@@ -724,6 +730,50 @@ def run(self):
                 show_error = False
             self.plot_num += 1
 
+    def add_segs(self, args):
+        """ If requested add DQ segments"""
+        std_segments = \
+            [
+                '{ifo}:DMT-GRD_ISC_LOCK_NOMINAL:1',
+                '{ifo}:DMT-DC_READOUT_LOCKED:1',
+                '{ifo}:DMT-CALIBRATED:1',
+                '{ifo}:DMT-ANALYSIS_READY:1'
+            ]
+        segments = list()
+        if hasattr(args, 'std_seg'):
+            if args.std_seg:
+                segments = std_segments
+            if args.seg:
+                for seg in args.seg:
+                    # NB: args.seg may be list of lists
+                    segments += seg
+
+            chan = args.chan[0][0]
+            m = re.match('^([A-Za-z][0-9]):', chan)
+            ifo = '?:'
+            if m:
+                ifo = m.group(1)
```

Comment:
this may be simplified as
```python
            ifo = m.group(1) if m else '?:'
```
but if you think that is less readable I'm not bothered.

Reply:
if the match fails m=None and m.<anything> throws an exception. N'est-ce pas?